Tchap/Modules/Application/AppCoordinator.swift

```diff
@@ -143,6 +220,103 @@ final class AppCoordinator: AppCoordinatorType {
+        // Check required parameters
+        guard let homeserver = registerParams["hs_url"],
+            let sessionId = registerParams["session_id"],
+            let clientSecret = registerParams["client_secret"],
+            let sid = registerParams["sid"] else {
+                NSLog("[AppCoordinator] handleRegisterAfterEmailValidation: failed, missing parameters")
+                return
+        }
+        
+        // Check whether there is already an active account
+        if self.mainSession != nil {
+            NSLog("[AppCoordinator] handleRegisterAfterEmailValidation: Prompt to logout current sessions to complete the registration")
+            AppDelegate.theDelegate().logout(withConfirmation: true) { (isLoggedOut) in
+                if isLoggedOut {
+                    self.handleRegisterAfterEmailValidation(registerParams)
+                }
+            }
+            return
+        }
+        
+        // Create a rest client
+        let restClientBuilder = RestClientBuilder()
+        restClientBuilder.build(homeserver) { (restClientBuilderResult) in
+            switch restClientBuilderResult {
+            case .success(let restClient):
+                if let identityServerURL = registerParams["is_url"] {
+                    restClient.identityServer = identityServerURL
+                }
+                
+                guard let identityServer = restClient.identityServer,
+                    let identityServerURL = URL(string: identityServer),
+                    let identityServerHost = identityServerURL.host else {
+                        let error = NSError(domain: MXKAuthErrorDomain, code: 0, userInfo: [NSLocalizedDescriptionKey: Bundle.mxk_localizedString(forKey: "error_common_message")])
+                        // FIXME: Present an error on coordinator.toPresentable()
+                        AppDelegate.theDelegate().showError(asAlert: error)
+                        return
+                }
+                
+                let registrationService = RegistrationService(accountManager: MXKAccountManager.shared(), restClient: restClient)
+                let deviceDisplayName = UIDevice.current.name
+                let threePIDCredentials = ThreePIDCredentials(clientSecret: clientSecret, sid: sid, identityServerHost: identityServerHost)
+                
+                registrationService.register(withEmailCredentials: threePIDCredentials, sessionId: sessionId, password: nil, deviceDisplayName: deviceDisplayName) { (registrationResult) in
+                    switch registrationResult {
+                    case .success:
+                        print("[AppCoordinator] handleRegisterAfterEmailValidation: success")
+                        _ = self.userDidLogin()
+                    case .failure(let error):
+                        // FIXME: Present an error on coordinator.toPresentable()
+                        AppDelegate.theDelegate().showError(asAlert: error)
```

Comment:
Create a private method in AppCoordinator to wrap call of `AppDelegate.theDelegate().showError(asAlert: error)` as this should be removed in the future.

Reply:
done